circuits/cpp/src/aztec3/utils/array.hpp

```diff
@@ -115,6 +115,30 @@ void array_push(Composer& composer, std::array<T, SIZE>& arr, T const& value)
     composer.do_assert(false, "array_push cannot push to a full array", CircuitErrorCode::ARRAY_OVERFLOW);
 };
 
+/**
+ * @brief Helper method to move all non-zero elements to the left of the array. E.g., [0,7,4,0,5] --> [7,4,5,0,0]
+ *        Remark: The ordering of the non-zero values is preserved.
+ *
+ * @tparam The type of the value stored in the array
+ * @tparam The size of the array
+ * @param The array into which we want to store the value
+ */
+template <typename T, size_t SIZE> void array_rearrange(std::array<T, SIZE>& arr)
```

Comment:
Add a new task to test these helpers.

Reply:
https://github.com/AztecProtocol/aztec-packages/issues/891